### PR TITLE
[AC-9445] Add IndustryCluster model

### PR DESCRIPTION
**Pull request by AmosWels**
## [AC-9445](https://masschallenge.atlassian.net/browse/AC-9445)

**Changes introduced**
- add the IndustryCluster model

**Testing**
- Checkout branch `AC-9445` and run `make migrate` on the accelerate repo,
- Confirm that the def of done according to [AC-9445](https://masschallenge.atlassian.net/browse/AC-9445) is met

[Sibling PR](https://github.com/masschallenge/accelerate/pull/3251)

### CODE REVIEW

File: accelerator/models/industry_cluster.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from __future__ import unicode_literals
```
Comment (iankigen) [inline accelerator/models/industry_cluster.py:1]: no need for unicode_literals

File: accelerator/models/industry_cluster.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import swapper
+
+from accelerator_abstract.models import BaseIndustryCluster
+
+
+class IndustryCluster(BaseIndustryCluster):
+    class Meta(BaseIndustryCluster.Meta):
+        swappable = swapper.swappable_setting(
+            BaseIndustryCluster.Meta.app_label,
+            "IndustryCluster")
```
Comment (iankigen) [inline accelerator/models/industry_cluster.py:12]: no need to use swapper

**Comment by iankigen** [comment]
we have some lint issues

File: accelerator/tests/factories/industry_cluster_factory.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from accelerator.models import IndustryCluster
+
+
+class IndustryClusterFactory(DjangoModelFactory):
+    class Meta:
+        model = IndustryCluster
+
+    name = Sequence(lambda n: "Climate Cluster {0}".format(n))
+    description = Sequence(
+        lambda n: "Program interested in Climate {0}".format(n))
```
Comment (iankigen) [inline accelerator/tests/factories/industry_cluster_factory.py:14]: Probably should be "Industry Cluster {}" instead of climate cluster

**Review by iankigen — COMMENTED** [review]
Looks good

**Review by iankigen — APPROVED** [review]

**Review by shakirandagire — APPROVED** [review]
QA looks good to me
 ✅ Model is created as per the definition of done and its migration exists
 ✅ IndustryCluster model is added to the admin site
Screenshot 2022-04-21 at 14 45 30
✅ Add IndustryClusterFactory with sensible default values
✅ add M2M relationship to:
CoreProfile - industry_cluster_interest
Program - supported_industry_clusters
✅ supported_industry_clusters option is added to the [Program administration site](http://staging.masschallenge.org/admin/mc/program/
Screenshot 2022-04-21 at 14 51 32
)

File: accelerator/tests/factories/core_profile_factory.py [hunk 4]
```diff
@@ -100,3 +100,18 @@ def industry_cluster_interest(self, create, extracted, **kwargs):
         if extracted:
             for cluster in extracted:
                 self.industry_cluster_interest.add(cluster)
+                
+    def program_location_interest(self, create, extracted, **kwargs):
```
Comment (iankigen) [inline accelerator/tests/factories/core_profile_factory.py:104]: needs `@post_generation` decorator

File: accelerator_abstract/models/base_industry_cluster.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    name = CharField(max_length=255)
+    description = TextField()
+    image = ImageField(
+        upload_to='industry_cluster_images',
+        blank=True, null=True)
+
+    class Meta(AcceleratorModel.Meta):
+        verbose_name_plural = 'Industry Clusters'
+        db_table = 'accelerator_industrycluster'
+        abstract = True
```
Comment (iankigen) [inline accelerator_abstract/models/base_industry_cluster.py:20]: this does not match what exists on the migration file

**Review by iankigen — CHANGES_REQUESTED** [review]
Inconsistent model and migration

**Review by iankigen — APPROVED** [review]

**Pull request merged**